addons/sale/models/sale_order_line.py

```diff
@@ -652,6 +659,38 @@ def _compute_qty_delivered(self):
         for so_line in lines_by_analytic:
             so_line.qty_delivered = mapping.get(so_line.id or so_line._origin.id, 0.0)
 
+    def _downpayment_state(self):
+        """
+        Check that this sale order line is a draft down payment
+        Note: self.ensure_one()
+        :return: True Verify that this SO line is a draft down payment
+        :rtype: Boolean
+        """
+        self.ensure_one()
+        if not self.is_downpayment:
+            return ""
+
+        if all(line.move_id.state == 'draft' for line in self._get_invoice_lines()
+               if line.move_id.amount_total == self.price_unit):
+            return _(' (Draft) ')
+        if all(line.move_id.state == 'cancel' for line in self._get_invoice_lines()
+               if line.move_id.amount_total == self.price_unit):
+            return _(' (Cancelled) ')
+
+        return ""
+
+    def _show_downpayment_section(self):
+        self.ensure_one()
+
+        if (
+                self.display_type == 'line_section'
+                and self.name == _("Down Payments")
```

Comment:
Never use a name/description/... as a mean of comparison.
It's not a constant value, might have been modified manually, or the language in the context is not the same as when the value was computed.